File: codeAccumulate/sparkStreaming/src/main/scala/com/sunsky/jdbcDataSource/DBUtil.scala
package com.sunsky.jdbcDataSource

import java.io.{File, FileInputStream, InputStream}
import java.sql.Connection
import java.util.Properties

import org.apache.commons.dbcp.BasicDataSource
import org.apache.spark.SparkFiles

/**
  * only for spark
  * use datasource to get jdbc connection
  */
object DBUtil {

  @volatile private var dataSource:BasicDataSource = null

  /**
    * init data source
    */
  private def initDataSource() : Unit = {

    try{

      val properties = new Properties()
      properties.load(new FileInputStream(new File(SparkFiles.get("db.properties"))))

      dataSource = new BasicDataSource
      dataSource.setDriverClassName("com.mysql.jdbc.Driver")
      dataSource.setUrl("jdbc:mysql://localhost:3306/bigdata?useUnicode=true&characterEncoding=utf-8")
      dataSource.setUsername("root")
      dataSource.setPassword("sunsky667")
      dataSource.setInitialSize(5)
      dataSource.setMaxActive(10)
    }catch {
      case e:Exception => e.printStackTrace()
    }

  }

  /**
    * get connection from data source
    * @return Connection
    */
  def getConnection() : Connection = {
    synchronized({
      if(dataSource == null){
        initDataSource()
      }
    })
    dataSource.getConnection
  }

  /**
    * close connection
    * @param connection
    */
  def close(connection: Connection) : Unit = {
    if(connection != null){
      try {
        connection.close()
      }catch {
        case e:Exception => e.printStackTrace()
      }
    }
  }

}

File: codeAccumulate/mariadb/src/main/java/com/sunsky/mariadb/DBUtil.java
package com.sunsky.mariadb;


import org.apache.commons.dbcp.BasicDataSource;

import java.sql.Connection;
import java.sql.SQLException;
import java.util.Properties;

public class DBUtil {

    private static BasicDataSource basicDataSouce;

    static {
        Properties properties = new Properties();
        try {

            properties.load(DBUtil.class.getClassLoader().getResourceAsStream("dbconfig.properties"));
            String driver = properties.getProperty("db.driver");
            String url = properties.getProperty("db.url");
            String user = properties.getProperty("db.user");
            String passwd = properties.getProperty("db.password");
            String intSize = properties.getProperty("db.intsize");
            String maxSize = properties.getProperty("db.maxsize");

            basicDataSouce = new BasicDataSource();
            basicDataSouce.setDriverClassName(driver);
            basicDataSouce.setUrl(url);
            basicDataSouce.setUsername(user);
            basicDataSouce.setPassword(passwd);
            basicDataSouce.setInitialSize(Integer.parseInt(intSize));
            basicDataSouce.setMaxActive(Integer.parseInt(maxSize));
        }catch (Exception e){
            e.printStackTrace();
        }
    }

    public static Connection getCon() throws SQLException{
        return  basicDataSouce.getConnection();
    }

    public static void closeCon(Connection connection) {
        try {
            if(connection != null){
                connection.close();
            }
        }catch (Exception e){
            e.printStackTrace();
        }
    }
}
